File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int numOfPeople = people.length;
        int answer = 0;
        int index=0;
        boolean[] visited = new boolean[numOfPeople];
        
        Arrays.sort(people);
        
        for(int i=numOfPeople - 1; i>=0; i--){
            if(visited[i]){
                continue;
            }
            else if(people[i] + people[index] <= limit){
                visited[i] = true;
                visited[index] = true;
                index++;
                answer++;
            }
            else{
                answer ++;
                visited[i] = true;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    n = len(people)
    answer = 0
    visited = [False] * n
    
    people.sort()
    left, right = 0, n-1
    
    while left <= right:
        if people[left] + people[right] <= limit:
            answer += 1
            left += 1
            right -= 1
        else:
            answer += 1
            right -= 1
    return answer